WalletWasabi.Fluent/ViewModels/Wallets/Home/History/HistoryItems/CoinJoinsHistoryItemViewModel.cs

```diff
@@ -87,6 +87,14 @@ private void Refresh()
 		IsConfirmed = CoinJoinTransactions.All(x => x.IsConfirmed());
 		Date = CoinJoinTransactions.Select(tx => tx.DateTime).Max().ToLocalTime();
 		OutgoingAmount = CoinJoinTransactions.Sum(x => x.Amount) * -1;
+
+		// OutgoingAmount/IncomingAmount should never be negative. In case of negative OutgoingAmount we gained money, for example: coordinator wallet in CJs.
+		if (OutgoingAmount < Money.Zero)
+		{
+			IncomingAmount = CoinJoinTransactions.Sum(x => x.Amount);
+			OutgoingAmount = null;
+		}
```

Comment:
Ahh nice catch! It is indeed a bug... 

cACK, but a better solution would be to create a new method in `HistoryItemViewModelBase.cs` called `SetAmount()` which has a `Money` parameter. And the method basically does what is in the `TransactionHistoryItemViewModel.cs` ctor:

```c#
if (amount < Money.Zero)
{
	OutgoingAmount = amount * -1;
}
else
{
	IncomingAmount = amount;
}
```

Once this method is ready, it should be called in the `TransactionHistoryItemViewModel.cs` ctor and in the `CoinJoinsHistoryItemViewModel`'s `Refresh()` method.

Reply:
You are right, this is more robust.
Done in https://github.com/zkSNACKs/WalletWasabi/pull/9618/commits/ce25dacc7f1890513621cc97d352ffb8264a0670